Gracefully handle missing post history on new drafts

niranjan-uma-shankar opened the pull request:
Fixes [#24515](https://github.com/TryGhost/Ghost/issues/24515), where clicking "Post history" on a newly created post (with no revision history) causes an uncaught exception, resulting in the screen hanging.

~This PR adds a guard to detect the absence of revision data and instead shows a user-friendly message~. This PR hides the "Post history" link in the settings panel for new posts. 

### BEFORE

https://github.com/user-attachments/assets/2e58a78e-107b-40ad-a545-070d58b2c502


### AFTER


https://github.com/user-attachments/assets/6b9107a6-442c-4ffa-b562-7d3e85f40f10

niranjan-uma-shankar committed:
Handles an uncaught exception when viewing Post history on a new post

This fixes an issue that causes the page to hang due to an exception when tapping "Post history" on a new post.

niranjan-uma-shankar committed:
Updates conditional to handle empty arrays in revisionList

This handles the case when revisionList is [], thereby still showing the empty state message.

niranjan-uma-shankar committed:
Guards selectedRevision.lexical before use

Ensures that this.selectedRevision.lexical will not throw an exception if selectedRevision is null/undefined.

niranjan-uma-shankar commented on ghost/admin/app/components/modal-post-history.js:
```diff
@@ -40,7 +40,7 @@ export default class ModalPostHistory extends Component {
 
     get currentTitle() {
-        return this.selectedRevision.title || this.post.get('title');
+        return this.selectedRevision?.title || this.post.get('title');
```
Fixed in [22d5291](https://github.com/TryGhost/Ghost/pull/24514/commits/22d52918f7f2167e5ccafacd761582b268597362).

kevinansfield commented:
@niranjan-uma-shankar thanks for looking at this! However, I think this approach is a bit over-complicated. The simpler solution would be not to show the Post History link in the settings menu when the client-side post model has the `isNew` property.

niranjan-uma-shankar commented:
@kevinansfield Thanks for reviewing!

> The simpler solution would be not to show the Post History link in the settings menu

That makes sense. My thinking was that showing the "Post History" link, regardless of whether there’s any history yet, might help new users discover the feature and understand where to find it later. It felt like a reasonable trade-off for a few additional lines of code, especially since it avoids conditionally hiding UI.

That being said, the scenario handled by this PR is an edge case. If the preference is to hide the link entirely for empty posts, I'm happy to update the PR to use the `isNew` check instead. Let me know what you'd prefer!

niranjan-uma-shankar committed:
Refactors the post revision history exception with a conditional visibility check

This replaces the empty revision history message with a check for new posts. Post history link is not displayed in the publish settings panel for new posts.
